R/util.R

```diff
@@ -350,3 +350,26 @@ show_latex_error <- function(file) {
     stop(e, ' See ', logfile, ' for more info.', call. = FALSE)
   }
 }
+
+# import two helper functions from knitr (with fallbacks if the knitr version is
+# not sufficient)
+is_html_output <- function(...) {
+  fun <- tryCatch(getFromNamespace('is_html_output', 'knitr'), error = function(e) {
```

Comment:
I think it would be cleaner if you switched behaviours based on `packageVersion("knitr")`. When you use a `tryCatch()` in this way, I worry about other errors being silently ignored.